lib/ts/I_Layout.h

```diff
@@ -31,80 +31,93 @@
 #ifndef _I_Layout_h
 #define _I_Layout_h
 
+#include <string>
+#include "ts/string_view.h"
+
 /**
   The Layout is a simple place holder for the distribution layout.
 
  */
 struct Layout {
-  Layout(const char *prefix = 0);
+  Layout(ts::string_view prefix = "");
   ~Layout();
 
   /**
    Return file path relative to Layout->prefix
    Memory is allocated, so use ats_free() when no longer needed
 
   */
-  char *relative(const char *file);
+  char *relative(ts::string_view file);
```

Comment:
I agree this is what we would want to have. We need to overload this at the moment, or change all calls to this function to deal with a std::string copy being returned. I believe this is why we have the function below